tqtezos/tezos-k8s
# Multinode test network with configurable number of nodes. #17

Based on aryeh/test-chain branch

Pod names have been changed as follows for clarity:

* the baking node is now called "tezos-progenitor"
* the non-baking nodes are called simply "tezos-nodes"

A new parameter `--number-of-nodes N` lets you choose
how many nodes should be in your test topology.

You can scale up or down the topology by reissuing the same
mkchain command with a different number of nodes, then pipe
to apply to an existing cluster.

Here is an example output of `get pods` with 3 nodes:

```
NAME                                READY   STATUS      RESTARTS   AGE
activate-job-4ks45                  0/1     Completed   0          50s
tezos-node-0                        1/1     Running     0          49s
tezos-node-1                        1/1     Running     0          49s
tezos-progenitor-77c6c87586-5gg7h   2/2     Running     0          49s
```

The --number-of-nodes parameter is also supported for "invite chains".
It is possible to create a full mesh of nodes connecting to a
progenitor in another cluster.

The yaml template files have been reorganized:

* progenitor.yaml contains progenitor deployment config
* node.yaml contains node statefulset config

Other changes:

* give zerotier its own persistent volume to save its state
(it used to go to the node storage)
* remove some commented out code
* remove stale pv.yaml file
